Windows build failures on some machines

```
[1/7] Performing configure step for 'llvm'
FAILED: llvm/src/llvm-stamp/llvm-configure
cmd.exe /C "cd /D C:\dev\gapid\out\release\aarch64\src\aarch64-build\llvm\src\llvm-build && c:\dev\cmake\bin\cmake.exe -CC:/dev/gapid/out/release/aarch64/src/aarch64-build/llvm/tmp/llvm-cache-Release.cmake -GNinja C:/dev/gapid/src/github.com/google/gapid/third_party/llvm && c:\dev\cmake\bin\cmake.exe -E touch C:/dev/gapid/out/release/aarch64/src/aarch64-build/llvm/src/llvm-stamp/llvm-configure"
loading initial cache file C:/dev/gapid/out/release/aarch64/src/aarch64-build/llvm/tmp/llvm-cache-Release.cmake
CMake Error at cmake/modules/GetHostTriple.cmake:24 (message):
  Failed to execute
  C:/dev/gapid/src/github.com/google/gapid/third_party/llvm/cmake/config.guess
Call Stack (most recent call first):
  cmake/config-ix.cmake:344 (get_host_triple)
  CMakeLists.txt:582 (include)
```

--- a/cmd/do/build.go
+++ b/cmd/do/build.go
@@ -21,7 +21,6 @@ import (
 	"strings"
 
 	"github.com/google/gapid/core/log"
-	"github.com/google/gapid/core/os/shell"
 )
 
 func doGlob(ctx log.Context, cfg Config) {
@@ -57,10 +56,7 @@ func doBuild(ctx log.Context, cfg Config, options BuildOptions, targets ...strin
 }
 
 func doCMake(ctx log.Context, cfg Config, options BuildOptions, targets ...string) {
-	env := shell.CloneEnv().AddPathStart("PATH",
-		cfg.bin().System(),
-		cfg.JavaHome.Join("bin").System(),
-	)
+	env := env(cfg)
 	args := []string{
 		"-GNinja",
 		"-DCMAKE_MAKE_PROGRAM=" + cfg.NinjaPath.Slash(),
@@ -96,13 +92,7 @@ func doCMake(ctx log.Context, cfg Config, options BuildOptions, targets ...strin
 }
 
 func doNinja(ctx log.Context, cfg Config, options BuildOptions, targets ...string) {
-	env := shell.CloneEnv().AddPathStart("PATH",
-		cfg.bin().System(),
-		cfg.JavaHome.Join("bin").System(),
-	)
-	if !cfg.MSYS2Path.IsEmpty() {
-		env.AddPathStart("PATH", cfg.MSYS2Path.Join("mingw64/bin").System()) // Required to pick up DLLs
-	}
+	env := env(cfg)
 	args := targets
 	if options.DryRun {
 		args = append([]string{"-n"}, args...)

--- a/cmd/do/env.go
+++ b/cmd/do/env.go
@@ -1,0 +1,54 @@
+// Copyright (C) 2017 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"runtime"
+
+	"github.com/google/gapid/core/os/file"
+	"github.com/google/gapid/core/os/shell"
+)
+
+func env(cfg Config) *shell.Env {
+	path := []string{}
+
+	if runtime.GOOS == "windows" {
+		if !cfg.MSYS2Path.IsEmpty() {
+			path = append(path,
+				cfg.MSYS2Path.Join("usr/bin").System(),     // Required for sh.exe and other unixy tools.
+				cfg.MSYS2Path.Join("mingw64/bin").System(), // Required to pick up DLLs
+			)
+		}
+
+		// Add windows and system32 to path
+		cmd, err := file.FindExecutable("cmd.exe")
+		if err != nil {
+			panic("Couldn't find cmd.exe on PATH")
+		}
+		system32 := cmd.Parent()
+		windows := system32.Parent()
+		path = append(path, system32.System(), windows.System())
+	}
+
+	path = append(path,
+		cfg.bin().System(),
+		cfg.JavaHome.Join("bin").System(),
+	)
+
+	env := shell.CloneEnv()
+	env.Unset("PATH")
+	env.AddPathEnd("PATH", path...)
+	return env
+}

--- a/cmd/do/env.go
+++ b/cmd/do/env.go
@@ -40,6 +40,19 @@ func env(cfg Config) *shell.Env {
 		system32 := cmd.Parent()
 		windows := system32.Parent()
 		path = append(path, system32.System(), windows.System())
+	} else {
+		added := map[file.Path]bool{}
+		for _, name := range []string{"sh", "uname", "sed", "clang", "gcc", "node"} {
+			exe, err := file.FindExecutable(name)
+			if err != nil {
+				continue
+			}
+			dir := exe.Parent()
+			if !added[dir] {
+				path = append(path, dir.System())
+				added[dir] = true
+			}
+		}
 	}
 
 	path = append(path,

